### PR TITLE
Add container push crane

### PR DESCRIPTION
**Pull request by dvonessen**
What does this PR change? What problem does it solve?
-----------------------------------------------------

This PR adds a new job to the addons for pushing container images from a local tarball archive to a remote registry.

Was the change discussed in an issue?
------------------------------------------------------------

None

Checklist
---------

- [x] I have read the [Contribution Guidelines](https://github.com/dbsystel/gitlab-ci-python-library/blob/main/CONTRIBUTING.md)
- [x] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
- [x] I have added tests for all changes in this PR
- [x] I have added api documentation for the changes
- [x] I added a changelog entry to the unreleased section with the PR number. [Changelog](https://github.com/dbsystel/gitlab-ci-python-library/blob/main/CHANGELOG.md)
- [x] Linting is done, and linting jobs are successfull
- [x] I'm done, this Pull Request is ready for review

### CODE REVIEW

File: gcip/addons/container/jobs/crane.py [hunk 1]
```diff
@@ -48,3 +48,53 @@ def copy(
...
+            If `None` it defaults internally to `PredefinedVariables.CI_PROJECT_NAME`. Defaults to None.
+        docker_client_config (Optional[DockerClientConfig], optional): Creates the Docker configuration file base on objects settings,
+            used by crane to authenticate against given registries. Defaults to None.
+        crane_image (Optional[Union[Image, str]], optional): Container image which contains `crane` command.
+            Defaults to PredefindedImages.CRANE.
+
+    Returns:
+        Job: Returns a `gcip.Job`, with neccessary configuration to push a container image stored as a `tarball` to a remote registry.
+    """
+    if not crane_image:
```
Comment (ThomasSteinbach) [inline gcip/addons/container/jobs/crane.py:81]: Please set the default values in the signature of this method.
* You will spare six lines of code without necessary logic.
* It is much more obvious for users of the method what the default values of those method arguments are.

There is no need to be anxious about those values are then class wide available. Those values are not meant to be modified at object level so there should be no side effect between objects of this class.

**Review by ThomasSteinbach — CHANGES_REQUESTED** [review]
Beautiful contribution!

File: gcip/addons/container/jobs/crane.py (hunk 1 above)
Comment (dvonessen) [reply gcip/addons/container/jobs/crane.py]: Hell yeah, i did that. Thank you.

**Comment by ThomasSteinbach** [comment]
I mentioned one more open task: You have to monkeypatch some more environment variables:
https://gitlab.com/thomassteinbach/gitlab-ci-python-library/-/jobs/1156682502

**Pull request merged**